### ISSUE TITLE
FDM issues

### ISSUE THREAD

**Issue by jan-hudec** [FGMEMBERS/A320neo#7]
The FDM is really not very good.

The most obvious problem is that the engines generate way too much thrust at speed, because they use the default thrust function which is valid for turbojets with supersonic intake and not high-bypass turbofans. But that is also the easiest problem to fix; I already [asked](http://aviation.stackexchange.com/questions/19458/how-does-turbofan-engine-performance-depend-on-speed-and-density-altutude) what the table should look like and can create one that will still be a guess, but at least a sensible one.

However there are other problems. The second obvious issue is that the aircraft tends to cruise with nose about 2° _below_ horizon while in reality it should be about 2° _above_ horizon, indicating that the lift table needs some tweaking. The lift function is not the hard part either.

The hard bits are not immediately obvious when testing the model out, only by either reading the model or disabling the FBW and taking the aircraft out of the flight envelope:
- The moment of inertia tensor is diagonal and I don't really know how good the numbers are (calculating moments of inertia is _hard_).
- The lift table does not include values for negative angle of attack nor for deep stall.
- While A320 does tend to stall straight (which I suppose is due to the good yaw damper) there _should_ be _some_ wing drop tendency and it _should_ be possible to get the aircraft in a spin, especially when yaw damper is disabled, but AFAICT the model is not capable of simulating either.
- Aileron reversal in stall and stall of horizontal stabilizer with loss of elevator efficiency don't seem to be simulated either.
- There is no simulation of critical Mach number.

Now these problems should be actually fixable, but a really big amount of work.

On the other hand I already tried (for the old A320family) replacing the FDM with a Yasim one. This has the advantage that it calculates the inertia tensor, uses saneish lift and drag for all attitudes and angles of attack and simulates asymmetric stall.

But I was never really satisfied with that, because some of the deficiencies apply to Yasim and they are _not_ easily fixable there. Yasim also does not provide many properties JSBSim exports that are useful for the FBW though I didn't look at the current implementation how much it actually needs them.

So I am not sure how to best improve the situation. I am opening this as tracking issue and to collect ideas.

**Comment by xcvb85** [comment]
Calculating moments of inertia is not that hard (depending on the accuracy) but the question is how to get detailed data of the mass distribution. If you already have a Yasim FDM and if you are happy with the results you could use its inertia tensor for the JSBSim FDM as well. If the mixed terms such as Ixy are zero it means that your axis are principal axis.

**Comment by jan-hudec** [comment]
@xcvb85, indeed, it should be possible to copy the values calculated by Yasim except one would have to be careful about the different axis definitions. I'll probably do that. Or I'll copy the calculations from Yasim to a separate tool; there are _many_ JSBSim models for FlightGear that could use this kind of updates.

Regarding principal axis, well, the thing is that no aircraft has the normal longitudinal axis as principal axis of its moment of inertia at least because the vertical stabilizer only extends up.

**Comment by jan-hudec** [comment]
Ok, so I've started a tool for generating model bits, https://github.com/jan-hudec/yaamatic, and generated new engine configurations with it.

The climb rate is no longer so absurdly huge, but the drag at altitude seems to me to be a little bit on the low side and correspondingly the operating ceiling a bit high. A320s usually cruise relatively low (say, FL320 rather than, say, FL380) and while neo has slightly more powerful engines in the base configuration, I believe it is because with higher bypass ratio the thrust declines faster with speed, so at cruise they probably don't have higher thrust any more (unfortunately http://www.cfmaeroengines.com/ only gives approximate cruise thrust for CFM56s, but not CFM Leaps).

The other thing is that JSBSim (nor YASim for that matter) does not simulate the increase of TSFC at cruise, which I _suspect_, but don't have confirmed, is with speed and which is rather huge. http://www.jet-engine.net/civtfspec.html gives 0.330 lb/lbf/h at SLS and 0.596 lb/lbf/h at cruise for CFM56-5A1 (used on A320-211; no data for Leap yet, so I just deduce 15% from those numbers). I am contemplating whether to adjust the fuel amounts when moving it to the feed tanks, create my own model or contribute change to JSBSim that would allow full `<function>`s in the engine parameters. It is probably the least pressing problem of the FDM right now though.

**Comment by legoboyvdlp** [comment]
By the way, if you need manuals or anything, they are available on Google. Like, charts for MAX alt.

**Comment by pypdeveloper** [comment]
What is the issue with the FMC.
